### PR TITLE
Changed how the aws config is created to be inline with aws documenta…

### PR DESCRIPTION
**Pull request by bpotaczek**
Moved config into it's own func
Changed config to use LoadDefaultConfig function

### CODE REVIEW

File: pkg/controllers/genericissuer_controller.go [hunk 1]
```diff
@@ -153,3 +124,40 @@ func validateIssuer(spec *api.AWSPCAIssuerSpec) error {
 	}
 	return nil
 }
+
+func (r *GenericIssuerReconciler) getConfig(ctx context.Context, spec *api.AWSPCAIssuerSpec) (aws.Config, error) {
+	if spec.Region != "" {
+		return config.LoadDefaultConfig(ctx,
+			config.WithRegion("us-east-1"),
```
Comment (divyansh-gupta) [inline pkg/controllers/genericissuer_controller.go:131]: Will hard-coding `us-east-1` here break current users of this package who don't specify a region in their spec but don't run in default `us-east-1`?

**Comment by divyansh-gupta** [comment]
@bpotaczek Could you please fix the Linting and Unit Test errors?

File: pkg/controllers/genericissuer_controller.go [hunk 2]
```diff
@@ -153,3 +124,39 @@ func validateIssuer(spec *api.AWSPCAIssuerSpec) error {
...
+		if !ok {
+			return aws.Config{}, errNoSecretAccessKey
+		}
+
+		return config.LoadDefaultConfig(ctx,
+			config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(string(accessKey), string(secretKey), "")),
+		)
+	}
+
+	return aws.Config{}, nil
```
Comment (divyansh-gupta) [inline pkg/controllers/genericissuer_controller.go:161]: Should this line also be `config.LoadDefaultConfig`?

Comment (RedbackThomson) [reply pkg/controllers/genericissuer_controller.go:161]: Yes :)

**Review by anbaig — APPROVED** [review]

**Comment by divyansh-gupta** [comment]
I have tested the change with IRSA, looks good to me. 🚀

**Review by divyansh-gupta — APPROVED** [review]

**Pull request merged**